src/components/OSCALMetadata.js

```diff
@@ -33,8 +37,131 @@ const useStyles = makeStyles((theme) => ({
 
+function saveModifiedMetadata(modifiedField, modifiableMetadata, newValue) {
+  /* Since the browser url will be of the form
+   * http://localhost:8080/system-security-plans?url=http://localhost:8080/oscal/v1/ssps/{uuid}
+   * and we just want the URL defined in the url parameter of the browser url, we must find the
+   * location of the ? and increment that value by 5 to get the start of the desired url.
+   */
+  const startOfUrl = window.location.href.indexOf("?") + 5;
+  const url = window.location.href.substring(startOfUrl);
+
+  /* We must specify a method of PATCH so the backend service knows we are just
+   * updating a value.
+   *
+   * We also provide a body so we can specify what needs to be modified.
+   */
+  fetch(url, {
+    method: "PATCH",
+    body: JSON.stringify({
+      metadata: {
+        modifiedField: newValue,
+      },
+    }),
+  })
+    .then((res) => res.json())
+    .then(
+      (result) => {
+        /* If we reach this point, then the OSCAL file has successfully been modified
+         * and we want those changes to be visible to the user.
+         */
+        modifiableMetadata["last-modified"][1](
+          formatDate(result["last-modified"])
+        );
+        modifiableMetadata[modifiedField].value[1](newValue);
+      },
+      () => {
+        alert(
+          `Could not update the ${modifiedField} metadata field with value: ${newValue}.`
+        );
+      }
+    );
+}
+
+function displayEditableTextField(modifiedField, modifiableMetadata) {
+  return modifiableMetadata[modifiedField].edit[0] ? (
+    <Typography variant={modifiedField === "title" ? "h6" : "body2"}>
+      <TextField
+        size="medium"
+        variant="outlined"
+        inputRef={modifiableMetadata[modifiedField].ref}
+      >
+        {modifiableMetadata[modifiedField].value[0]}
+      </TextField>
+    </Typography>
+  ) : (
+    <Typography variant={modifiedField === "title" ? "h6" : "body2"}>
+      {modifiableMetadata[modifiedField].value[0]}
+    </Typography>
+  );
+}
+
+function displayEditIcons(modifiedField, modifiableMetadata) {
+  const edit = modifiableMetadata[modifiedField].edit[0];
+  const setEdit = modifiableMetadata[modifiedField].edit[1];
+
+  return edit ? (
+    <>
+      <IconButton
+        onClick={() => {
+          setEdit(!edit);
+        }}
+      >
+        <CloseIcon fontSize="small" />
+      </IconButton>
+      <IconButton
+        onClick={() => {
+          const newValue = modifiableMetadata[modifiedField].ref.current.value;
+          saveModifiedMetadata(modifiedField, modifiableMetadata, newValue);
+          setEdit(!edit);
+        }}
+      >
+        <SaveIcon fontSize="small" />
+      </IconButton>
+    </>
+  ) : (
+    <IconButton
+      onClick={() => {
+        setEdit(!edit);
+      }}
+    >
+      <EditIcon fontSize="small" />
+    </IconButton>
+  );
+}
+
 export default function OSCALControlGuidance(props) {
```

Comment:
Is there any reason this is named `OSCALControlGuidance` and not `OSCALMetadata`?

Reply:
Looks like a leftover from a copy/paste to me.